source/jsonifiable.d.ts

```diff
@@ -0,0 +1,32 @@
+import type {JsonPrimitive, JsonValue} from './basic';
+
+type JsonifiableObject = {[Key in string]?: Jsonifiable} | {toJSON: () => JsonValue};
+type JsonifiableArray = Jsonifiable[];
+
+/**
+Matches a value that can be losslessly converted to JSON.
+
+`undefined` is allowed in object fields (ex: `{a?: number}`) as a
+special case even though `JSON.stringify({a: undefined})` is `{}`.
```

Comment:
Don't hard-wrap.

It think the user will ask themselves why `undefined` is allowed, so I think the docs here should quickly answer that.